httpconnect/httpconnect.go

```diff
@@ -147,28 +151,37 @@ func (f *HTTPConnectHandler) intercept(w http.ResponseWriter, req *http.Request)
 			}
 		}
 	}
-	var closeOnce sync.Once
-	go func() {
+
+	readErrCh := make(chan error)
+	op.Go(func() {
```

Comment:
It allows the op's context to propagate, which achieves what you said but also more.  In particular, if this goroutine calls ops.Enter(), that will be a child of the outer op.